Common/Product/SharedProject/ProjectNode.CopyPaste.cs

```diff
@@ -786,45 +799,51 @@ class FolderAddition : Addition {
                     var sourceFolder = Project.FindNodeByFullPath(SourceFolder) as FolderNode;
                     if (sourceFolder == null || DropEffect != DropEffect.Move) {
                         newNode = Project.CreateFolderNodes(NewFolderPath);
+
+                        foreach (var addition in Additions) {
+                            addition.DoAddition(ref overwrite);
+                        }
+
+                        if (sourceFolder != null) {
+                            if (sourceFolder.IsNonMemberItem) {
+                                // copying or moving an existing excluded folder, new folder
+                                // is excluded too.
+                                ErrorHandler.ThrowOnFailure(newNode.ExcludeFromProjectWithRefresh());
+                            } else if (sourceFolder.Parent.IsNonMemberItem) {
+                                // We've moved an included folder to a show all files folder,
+                                //     add the parent to the project   
+                                ErrorHandler.ThrowOnFailure(sourceFolder.Parent.IncludeInProjectWithRefresh(false));
+                            }
+
+                            if (DropEffect == DropEffect.Move) {
```

Comment:
How can we ever get here? The nested if conditions are `(sourceFolder == null || DropEffect != Move) && sourceFolder != null && DropEffect == Move`

Reply:
Yeah you're right, it's not needed.